src/opentimelineio/stackAlgorithm.cpp

```diff
@@ -168,14 +215,28 @@ flatten_stack(Stack* in_stack, ErrorStatus* error_status)
 Track*
 flatten_stack(std::vector<Track*> const& tracks, ErrorStatus* error_status)
 {
+    std::vector<Track*> flat_tracks;
+    TrackRetainerVector tracks_retainer;
+    flat_tracks.reserve(tracks.size());
+    for (auto track : tracks)
+    {
+        flat_tracks.push_back(track);
+    }
```

Comment:
Could you put a comment above this tracks_retainer section with a clue to future readers about why the retainer is needed? I _think_ I understand, but I had to stare at it for a while :)